### PR TITLE
feat(client): real unilateral exit (unroll) with tree PSBT signing

### PR DESCRIPTION
**Pull request by lobbyclawy**
## Issue #398

Implements real unilateral exit (unroll) with tree PSBT signing in the Rust client.

### Changes

**`crates/dark-client/src/client.rs`:**
- Updated `unroll(pubkey)` to implement the full unilateral exit flow:
  1. Lists spendable VTXOs via `GetVtxos`
  2. For each VTXO, queries `GetVtxoChain` to get the chain (commitment → tree → leaf)
  3. Walks backwards to find the next offchain tree transaction to broadcast
  4. Fetches pre-signed PSBTs via `GetVirtualTxs`
  5. Finalizes each PSBT (taproot key-path and script-path spends)
  6. Returns raw tx hex strings for broadcasting

- Added `finalize_tree_psbt()` — finalizes pre-signed tree PSBTs by constructing witness data from `tap_key_sig` (key-path) or `tap_scripts`/`tap_script_sigs` (script-path)

- Replaced stub `RedeemBranch` with real implementation that queries the indexer's `GetVtxoChain` RPC

**`tests/e2e_regtest.rs`:**
- Updated `test_unilateral_exit_leaf_vtxo` — settles VTXOs, unrolls, broadcasts, verifies on-chain
- Updated `test_unilateral_exit_preconfirmed_vtxo` — multi-level unroll for preconfirmed VTXOs
- Updated `test_sweep_checkpoint` — unroll + sweep after expiry
- Added `broadcast_tx_hex()` helper for broadcasting via Esplora

**Note:** `unroll()` signature changed to `unroll(pubkey: &str)` — all callers updated.

### Reference
- Go unroll: `vendor/arkd/pkg/client-lib/unroll.go`
- Go redemption: `vendor/arkd/pkg/client-lib/redemption/redeem.go`

### CODE REVIEW

**Review by lobbyclawy — COMMENTED** [review]
## Review Summary

Solid implementation replacing the stub `unroll()` with a real VTXO tree PSBT signing flow. The approach correctly uses GetVtxoChain + GetVirtualTxs to fetch and finalize tree PSBTs. E2E tests updated accordingly. A few observations on edge cases and code patterns below.

File: crates/dark-client/src/client.rs [hunk 1]
```diff
@@ -778,74 +779,171 @@ impl ArkClient {
...
+            }
+        }
+
+        if next_txids.is_empty() {
+            return Ok(vec![]);
+        }
+
+        // 3. Fetch PSBTs for the next offchain txids
+        let indexer = self
+            .indexer
```
[the remark this answers is not in the document]
Comment (lobbyclawy) [reply crates/dark-client/src/client.rs]: Agreed the borrow lifecycle is correct since the loop completes before line 813. The current structure is clear enough — extracting a helper would add indirection without benefit here since `RedeemBranch::new` needs `&mut Option<Client>` directly. No change needed.

File: crates/dark-client/src/client.rs [hunk 2]
```diff
@@ -778,74 +779,171 @@ impl ArkClient {
...
+            input.tap_key_sig = None;
+            input.tap_internal_key = None;
+            input.tap_merkle_root = None;
+            input.tap_scripts.clear();
+            input.tap_script_sigs.clear();
+            input.witness_utxo = None;
+        } else {
+            return Err(ClientError::Serialization(format!(
+                "Input {idx}: no taproot key-spend sig or leaf script"
+            )));
```
[the remark this answers is not in the document]
Comment (lobbyclawy) [reply crates/dark-client/src/client.rs]: Correct — when all entries are Commitment/Unspecified, `next_offchain_txid` stays `None` and the VTXO is simply skipped in the unroll loop (line 805 only adds to `next_txids` if `Some`). This is the intended behavior for VTXOs that have no offchain tree txs to broadcast. A trace-level log could be useful for debugging but would add noise in the happy path — acceptable as-is.
